src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java

```diff
@@ -490,8 +490,8 @@ private void internalFlush(LinkedBlockingQueue<String> buffer, String format,
                   "Error sending " + entityType + " to Wavefront (HTTP " + statusCode + "). Data " +
                       "will be requeued and resent.");
               requeue(buffer, items, dropped, entityType, bufferFullMessageType);
-            }
           }
+        }
```

Comment:
`if (400 <= statusCode && statusCode <= 599)` condition should be updated to include -1 in order for points to be kepted and requeued.  Same for `WavefrontDirectIngestionClient`.